torch/csrc/jit/import.cpp

```diff
@@ -319,4 +320,113 @@ std::shared_ptr<script::Module> ImportIRModule(
   return decoder.decode(module, serialized_module, storage_map);
 }
 
+// TODO: mmap file reader
+class PyTorchFileReader {
+ public:
+  PyTorchFileReader(const std::string& filename) {
+   fp = std::fopen(filename.c_str(), "rb");
+   std::fseek(fp, 0L, SEEK_END);
+   file_size = std::ftell(fp);
+   std::fseek(fp, 0L, SEEK_SET);
+   readAndValidateFileHeader();
+  }
+
+  // returns raw data map and the index of the last record (i.e. the ModelProto)
```

Comment:
This isn't the API we eventually want. This PR should strive to get the API right for the reader or writer classes even if the implementation may change. Right now the API is weirdly designed to match up with the ModelDecoder API. We should be changing the API of both to create clear abstraction boundaries.